android/src/main/java/io/mosip/tuvali/ble/central/impl/Controller.kt

```diff
@@ -92,8 +99,25 @@ class Controller(val context: Context) {
     gattClient?.discoverServices(this::onServicesDiscovered, this::onServiceDiscoveryFailure)
   }
 
-  fun requestMTU(mtu: Int) {
-    gattClient?.requestMtu(mtu, this::onRequestMTUSuccess, this::onRequestMTUFailure)
+  fun requestMTU(mtuValues: Array<Int>, delayTime: Long) {
+    this.mtuValues = mtuValues
+    while(requestMtuCounter<mtuValues.size) {
+      if (isMTURequestCallbackReceived) {
+        break
+      }
+      requestedMTUValue = mtuValues[requestMtuCounter++]
+      gattClient?.requestMtu(requestedMTUValue, this::onRequestMTUSuccess, this::onRequestMTUFailure)
+      sleepInRealTime(delayTime)
+    }
+    if(!isMTURequestCallbackReceived) {
+      onRequestMTUFailure(0)
+    }
```

Comment:
Should this be inside the while-loop or is `onRequestMTUFailure` executed when each attempt to exchange a common MTU fails?